### DISCUSSION TITLE
How to delete uploaded file on RichEditor?

### DISCUSSION THREAD

**Discussion by diazsasak** [filamentphp/filament#1871] [Help] [answered]
Hi, 

how to delete the uploaded file when we have upload an image but not proceed to save the form or when we delete the record?

Thank you

**Comment by titantwentyone** [comment, marked as answer]
The file is just temporary.I the record is not saved Livewire will clean itself up. When you delete a record, you have to clean up the file manually. Docs state that you can use Model events to perform this cleanup:

> Please note, it is the responsibility of the developer to delete these files from the disk if they are removed, as Filament is unaware if they are depended on elsewhere. One way to do this automatically is observing a [model event](https://laravel.com/docs/eloquent#events).

You can implement the following to remove the file when it is deleted from the control (i.e. click the "x" in the top right):

```
FileUpload::make('something')
    ->deleteUploadedFileUsing(function($file) {
        Storage::disk('public')->delete($file);
    });

**Reply by North-Gemini** [reply]
Thanks titantwentyone, your answer was really helpful

**Reply by diazsasak** [reply]
what about RichEditor input? same as FileUpload input?

**Reply by danharrin** [reply]
Use an observer for that
